### PR TITLE
Vellum app callout templates config file

### PR DESCRIPTION
**Pull request by millerdev**
Related PR (no hard dependency): https://github.com/dimagi/Vellum/pull/505

@emord 
cc @sravfeyn

### CODE REVIEW

File: corehq/apps/app_manager/views.py [hunk 1]
```diff
@@ -1272,10 +1273,32 @@ def form_designer(request, domain, app_id, module_id=None, form_id=None,
...
+    """
+    path = os.path.join(
+        os.path.dirname(__file__),
+        'static', 'app_manager', 'json', 'vellum-app-callout-templates.yaml'
+    )
+    if os.path.exists(path):
+        with open(path) as f:
+            data = yaml.load(f)
+    else:
+        log.warning("not found: %s", path)
```
Comment (dimagimon) [inline corehq/apps/app_manager/views.py]: F821 undefined name 'log'

File: corehq/apps/app_manager/static/app_manager/json/vellum-app-callout-templates.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#           arg: ""
+#       response:
+#           r1: x
+#           r2: y
+#           r3: z
+#   -
+#       name: Other App Name
+#       id: com.google.other.app.id
+#
+# "extra" and "response" dicts are optional
```
Comment (millerdev) [inline corehq/apps/app_manager/static/app_manager/json/vellum-app-callout-templates.yaml]: @wspride here's your yaml file

File: corehq/apps/app_manager/views.py [hunk 3]
```diff
@@ -1272,10 +1273,32 @@ def form_designer(request, domain, app_id, module_id=None, form_id=None,
...
+        'app_callout_templates': next(app_callout_templates),
     })
     return render(request, 'app_manager/form_designer.html', context)
 
 
+def app_callout_templates():
+    """Load app callout templates from config file on disk
+
+    Generator function defers file access until needed, acts like a
+    constant thereafter.
```
Comment (czue) [inline corehq/apps/app_manager/views.py]: cool pattern - never seen it before. 

is it right that the benefits of this are the lazy evaluation and fast / constant memory access, and the drawbacks are that if anyone ever tries to iterate over the whole thing they will create an infinite loop?

based on the failure mode I might advocate marking it "private" to the module with a leading `_`

**Comment by sravfeyn** [comment]
Looks good. g2m, once cory's comment is resolved

File: corehq/apps/app_manager/views.py (hunk 3 above)
Comment (millerdev) [inline corehq/apps/app_manager/views.py]: Yes and yes. Good call on making it private.

**Comment by millerdev** [comment]
:white_check_mark:

**Pull request merged**